### PR TITLE
Update sync functionality

### PR DESCRIPTION
**Pull request by CaillaudPA**
🚀 **PR: Enhanced Synchronization & Installation Procedures for AiSmartTalk Module**

This PR introduces several enhancements to the AiSmartTalk module, focusing on improving synchronization mechanisms and streamlining the installation process. Key changes include:

- ✨ Added support for forced synchronization via the `forceSync` parameter, enabling immediate syncing with external APIs.
- 🛠️ Installation now includes a step to add a `synch` field to the `product` table, enhancing data integrity and synchronization capabilities.
- 🔧 Refactored the button display logic to include both new product synchronization and full re-synchronization options.
- 🐞 Fixed minor issues and improved code readability and formatting.

🔍 **Technical Details:**
- Modified files: `aismarttalk.php` and `FutureAiApi.php`.
- Added `addSynchField()` method for database schema updates during installation.
- Enhanced `sync()` method to support optional forced synchronization.
- Improved API interaction in `FutureAiApi` with conditional property setting and batched product data processing.

👀 Please review the changes for potential impacts on module functionality and user experience. Feedback and further testing are welcome to ensure seamless integration and performance improvements.

### CODE REVIEW

File: modules/aismarttalk/aismarttalk.php [hunk 1]
```diff
@@ -151,11 +155,21 @@ public function displayBackOfficeIframe() {
         }
     }
 
-    public function install() {
-        return parent::install() &&
-               $this->registerHook('displayFooter');
+    public function install()
+    {
+        if (!parent::install() || !$this->registerHook('displayFooter') || !$this->addSynchField()) {
```
Comment (OdranHUSSON) [inline modules/aismarttalk/aismarttalk.php]: wondering what happen on module update ?

File: modules/aismarttalk/aismarttalk.php [hunk 2]
```diff
@@ -151,11 +155,21 @@ public function displayBackOfficeIframe() {
...
+            return false;
+        }
+        return true;
     }
 
+    private function addSynchField()
+    {
+        $sql = 'ALTER TABLE '._DB_PREFIX_.'product ADD COLUMN synch TINYINT(1) NOT NULL DEFAULT 0;';
```
Comment (OdranHUSSON) [inline modules/aismarttalk/aismarttalk.php]: I think we should use a relaly specific field to be sure no other pluggin uses it, like aismarttalk_sync

Comment (CaillaudPA) [reply modules/aismarttalk/aismarttalk.php]: done

File: modules/aismarttalk/aismarttalk.php (hunk 1 above)
Comment (CaillaudPA) [reply modules/aismarttalk/aismarttalk.php]: there is no module update in our situation 

On passe direct par un zip

**Pull request merged**